cpp/include/cudf/strings/string.cuh

```diff
@@ -105,6 +105,30 @@ __device__ bool is_float(string_view const& d_str)
   }
   return result;
 }
+
+inline __device__ string_view shuffle_xor(string_view var, uint32_t delta)
+{
+  const char* data = reinterpret_cast<const char*>(
+    __shfl_xor_sync(~0, reinterpret_cast<uintptr_t>(var.data()), delta));
+  auto size = __shfl_xor_sync(~0, var.size_bytes(), delta);
+  return string_view(data, size);
+}
+
+template <typename Operator>
+inline __device__ string_view warp_reduce(string_view extremum_value, Operator op)
```

Comment:
Why use this instead of `cub::WarpReduce`? If you're worried that you would lose out on the shfl optimizations? Fear not, [cub slices your type to perform shfl on it](https://github.com/NVIDIA/cub/blob/fef1b9c3b3d095ce9c1b51bec70d3f6d7cac11db/cub/util_ptx.cuh#L529-L544).

Reply:
That's good to know! I haven't seen any info on this in their docs.